### DISCUSSION TITLE
Notification center implementation

### DISCUSSION THREAD

**Discussion by LBCrion** [LBCrion/sfwbar#189] [Ideas]
I've done some groundwork on #183 and now have a basic implementation of the notification center module in my local branch. There are still quite a few rough edges and the interface is still changing, but the basics are working (i.e. displaying the notification, removing it after it expires, closing through UI or remote request and support for buttons/actions). 

What I'm struggling with is what form the UI should take. So far I have the usual popup with notification layout blatantly copied from swaync:
![20240524_07h53m54s_grim](https://github.com/LBCrion/sfwbar/assets/76477076/193eee3d-aa4d-4354-ae12-1f5f61238c10)

I think it's probably worth grouping notifications by an application or a category. I.e. display a list of applications and corresponding app_icons and upon clicking on any of them, have the section expand to show the notifications? Perhaps have the first application expanded by default? 

As usual, UI design isn't my strong suit, so any ideas are appreciated.

**Comment by ffs-akl** [comment]
The ❎ icon in the red circle can be misleading,  as the bottom two❎ indicate closed,  this ❎ icon indicates folded.
![swappy-20241017-234803](https://github.com/user-attachments/assets/561e90e4-aa5d-4077-a57a-573c91d5a229)

**Comment by ffs-akl** [comment]
I've changed the icon to collapse-all, which I think better reflects what the button is actually doing here
![swappy-20241019-181623](https://github.com/user-attachments/assets/c54891dd-c66e-4046-b142-a70a8ac07a1a)
